qt/scientific_interfaces/ISISReflectometry/GUI/Options/OptionsDialogPresenter.cpp

```diff
@@ -0,0 +1,87 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "OptionsDialogPresenter.h"
+#include "IOptionsDialogView.h"
+
+namespace MantidQt {
+namespace CustomInterfaces {
+namespace ISISReflectometry {
+
+/**
+ * Construct a new presenter with the given view
+ * @param view :: a handle to a view for this presenter
+ * @param model :: a handle to a model for this presenter
+ */
+OptionsDialogPresenter::OptionsDialogPresenter(
+    IOptionsDialogView *view, std::unique_ptr<IOptionsDialogModel> model)
+    : m_view(view), m_model(std::move(model)), m_mainWindowNotifyee() {}
+
+/** Allow options to be initialised once the main presenter
+    is constructed (necessary due to rounding impl)
+ */
+void OptionsDialogPresenter::notifyInitOptions() { initOptions(); }
+
+void OptionsDialogPresenter::notifyOptionsChanged() {
+  m_mainWindowNotifyee->optionsChanged();
+}
+
+/** Subscribe the view to this presenter */
+void OptionsDialogPresenter::notifySubscribeView() { m_view->subscribe(this); }
```

Comment:
Could `m_view->subscribe` just be called in the `OptionsDialogPresenter` constructor? Then we wouldn't need this function?

Reply:
I seem to recall having trouble with this previously - would it not fail to subscribe and cause an access violation since the presenter doesn't exist until it is constructed and so `this` would be a nullptr?

Reply:
Seems to be fine now 🤷‍♂

Reply:
The notification is also used in some of the tests, so I will make calls in the presenter constructor but leave the function in for now.

Reply:
Looks like it's only used in `testPresenterSubscribesToView` and I think you can just put the `EXPECT_CALL` before `makePresenter` to avoid needing it?